modules/nextflow/src/main/resources/nextflow/executor/command-run.txt

```diff
@@ -150,11 +159,17 @@ nxf_main() {
     export NXF_TASK_WORKDIR="$PWD"
     {{stage_cmd}}
 
+    ## Deactivate fast failure to allow uploading stdout and stderr files later
     set +e
     (set -o pipefail; (nxf_launch | tee {{stdout_file}}) 3>&1 1>&2 2>&3 | tee {{stderr_file}}) &
     pid=$!
     wait $pid || nxf_main_ret=$?
-    {{unstage_cmd}}
+    if [[ ${nxf_main_ret:=0} == 0 ]]; then
+        ## Data unstaging redirecting stdout and stderr with append mode
+        (set -e -o pipefail; ({{unstage_cmd}} | tee -a {{stdout_file}}) 3>&1 1>&2 2>&3 | tee -a {{stderr_file}})
+        nxf_unstage_ret=$?
```

Comment:
Was this tested which S3 uploads that are made in a parallel logic? see [here](https://github.com/nextflow-io/nextflow/blob/test_3711/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchFileCopyStrategy.groovy#L121-L127)

Reply:
I tested with azure, that is also doing the copies in parallel.

Reply:
I have just seen, I created a validation test using a fake aws command to produce errors in stage_out